cmd/acra-server/acra-server.go

```diff
@@ -440,6 +442,30 @@ func realMain() error {
 		os.Exit(1)
 	}
 
+	{
+		var httpAPIConnWrapper network.HTTPServerConnectionWrapper
+		var err error
+		if *httpAPIUseTLS {
+			if !*enableHTTPAPI {
+				log.WithField(logging.FieldKeyEventCode, logging.EventCodeGeneral).
+					Warningln("--http_api_use_tls is provided, but the HTTP API server is not configured. Use --http_api_enable to enable it.")
+			}
+			httpAPIConnWrapper, err = common.BuildHTTPAPIConnectionWrapper(tlsWrapper, []byte(*clientID))
+		} else {
+			if *enableHTTPAPI {
+				log.WithField(logging.FieldKeyEventCode, logging.EventCodeGeneral).
+					Warningln("HTTP API server is used without TLS. Consider using TLS whenever possible.")
+			}
+			httpAPIConnWrapper, err = common.BuildHTTPAPIConnectionWrapper(nil, []byte(*clientID))
```

Comment:
The `clientID` is `""` by default. I checked, but I also remember it from the logs of proxy decryptor. So, I guess it's not a problem

Reply:
Ok, but how to check if TLS wrapper is empty? :eyes:

Reply:
```
if *httpAPIUseTLS {
			if !*enableHTTPAPI {
				log.WithField(logging.FieldKeyEventCode, logging.EventCodeGeneral).
					Warningln("--http_api_tls_transport_enable is provided, but the HTTP API server is not configured. Use --http_api_enable to enable it.")
				os.Exit(1)
			}
                        if clientID == "" || !*tlsUseClientIDFromCertificate {
                               ////////////////
                               // ERROR
                              ///////////////
                        }
			httpAPIConnWrapper, err = common.BuildHTTPAPIConnectionWrapper(tlsWrapper, []byte(*clientID))
		}
```

in case with tls, we should check that clientID != "" or we use clientID from certificate. Other cases will be validated in the wrapper's constructor - https://github.com/G1gg1L3s/acra/blob/63cfa8bf76b116aa202945dcb4d70762fbfe9afd/network/tls_wrapper.go#L166